src/main/java/com/esri/samples/editing/edit_and_sync_features/EditAndSyncFeaturesSample.java

```diff
@@ -0,0 +1,465 @@
+    geodatabaseButton.setText("Syncing Geodatabase...");
+    geodatabaseButton.setDisable(true);
+
+    // create parameters for the sync task
+    SyncGeodatabaseParameters syncGeodatabaseParameters = new SyncGeodatabaseParameters();
+    syncGeodatabaseParameters.setSyncDirection(SyncGeodatabaseParameters.SyncDirection.BIDIRECTIONAL);
+    syncGeodatabaseParameters.setRollbackOnFailure(false);
+
+    // get the layer ID for each feature table in the geodatabase, then add to the sync job
+    for (GeodatabaseFeatureTable geodatabaseFeatureTable : geodatabase.getGeodatabaseFeatureTables()) {
+      long serviceLayerId = geodatabaseFeatureTable.getServiceLayerId();
+      SyncLayerOption syncLayerOption = new SyncLayerOption(serviceLayerId);
+      syncGeodatabaseParameters.getLayerOptions().add(syncLayerOption);
+    }
+
+    // create a Sync Job and start it.
+    final SyncGeodatabaseJob syncGeodatabaseJob = geodatabaseSyncTask.syncGeodatabase(syncGeodatabaseParameters, geodatabase);
+    syncGeodatabaseJob.start();
+
+    // show progress
+    showProgress(syncGeodatabaseJob);
+
+    // add a job done listener to the Sync Job
+    syncGeodatabaseJob.addJobDoneListener(() -> {
+      if (syncGeodatabaseJob.getStatus() == Job.Status.SUCCEEDED) {
+        displayMessage("Database Sync Complete", null);
+
+        // update button text to signal we are ready to edit
+        geodatabaseButton.setText("Geodatabase Ready");
+        geodatabaseButton.setDisable(true);
+      } else {
+        displayMessage("Database did not sync correctly!", syncGeodatabaseJob.getError().getMessage());
+      }
+    });
+  }
+
+  /**
+   * Updates the extent of the area marked with a red border to be used for geodatabase generation
+   */
+  private void updateGeodatabaseExtentEnvelope() {
+    if (map.getLoadStatus() == LoadStatus.LOADED) {
+      // get the upper left corner of the view
+      Point2D minScreenPoint = new Point2D(50, 50);
+      // get the lower right corner of the view
+      Point2D maxScreenPoint = new Point2D(mapView.getWidth() - 50, mapView.getHeight() - 50);
+      // convert the screen points to map points
+      Point minPoint = mapView.screenToLocation(minScreenPoint);
+      Point maxPoint = mapView.screenToLocation(maxScreenPoint);
+      // use these points to define and create an envelope
+      if (minPoint != null && maxPoint != null) {
+        Envelope geodatabaseExtentEnvelope = new Envelope(minPoint, maxPoint);
+        // update the graphics
+        geodatabaseExtentGraphics.setGeometry(geodatabaseExtentEnvelope);
+      }
+    }
+  }
+
+  /**
+   * Show a progress bar
+   *
+   * @param job the job to show progress for
+   */
+  private void showProgress(Job job) {
+
+    // show progress
+    job.addProgressChangedListener(() -> progressIndicator.setVisible(true));
+
+    // hide progress indicator on complete
+    job.addJobDoneListener(() -> {
+      if (job.getStatus() == Job.Status.SUCCEEDED) {
```

Comment:
Should we handle the case where the job fails - otherwise you'll be left with progress showing even though the job is finished.